File: codeforces/contests/22/a.cpp
#include <bits/stdc++.h>

using namespace std;

/**
 * The Way to Home
 * https://codeforces.com/gym/445680/A
 *
 * Time - O(n)
 * Space - O(1)
 */

int main()
{
    int n, d;
    string s;

    int res = 0, i = 0;

    ios::sync_with_stdio(false);
    cin.tie(0);

    cin >> n >> d;
    cin >> s;

    while (i < n) {
        int j = min(i + d, n - 1);
        while (s[j] != '1' && j > i) {
            --j;
        }

        if (s[j] == '1' && j > i) {
            ++res;
            i = j;
        } else {
            break;
        }
    }

    if (i == n - 1) {
        cout << res;
    } else {
        cout << -1;
    }

    cout << "\n";

    return 0;
}

File: codeforces/contests/22/a.py
"""
New Year Transportation 

https://codeforces.com/gym/444629/problem/A

Time - O(n)
Space - O(1)
"""

n, t = map(int, input().split())

a = list(map(int, input().split()))

res = "NO"

i = 0
while i < n - 1:
    if t == i + 1:
        res = "YES"
        break
    i += a[i]

if t == i + 1:
    res = "YES"

print(res)
